File: Signiture-Backend/kotlin/study/src/main/java/operator/Exam.java
package operator;

public class Exam {

    public Exam() {
        var name = "홍길동";   // 타입추론

        Integer num = 10;
        callMethod(num);

        // NullPointerException
        Integer nullNum = null;
        callMethod(nullNum);

        callMethod(null);
    }

    private void callMethod(Integer i) {
        if (i == null) {
            System.out.println("null  값 입니다.");
        } else {
            System.out.println(i);
        }

        var temp = (i == null) ? "null 값 입니다." : i;
        System.out.println(temp);
    }

    public static void main(String[] args) {
        new Exam();
    }
}

File: Signiture-Backend/kotlin/study/src/main/kotlin/operator/Exam.kt
package pcy.study.operator

fun main() {
    val num = 10;
    callFunction(num)

    val nullNum: Int? = null
    callFunction(nullNum)

    callFunction()
}

fun callFunction(i: Int? = 100) {
    // Elvis 연산자
    // ? << null이 올 수도 있다.
    // 변수?. << 변수가 null이 아닐 때
    // 변수?: << 변수가 null 일 때
    i?.let {
        println(i)
    } ?: run {
        println("null  값 입니다.")
    }

//    val temp = if(i == null) "null 값 입니다." else i
    val temp = i?: "null 값 입니다."
    println(temp)
}